Tests/Foundation/UserDefaults/UserDefaults+PropertyWrapperTests.swift

```diff
@@ -0,0 +1,211 @@
+//
+//  UserDefaults+PropertyWrapperTests.swift
+//  Tests
+//
+//  Created by Jasmin Abou Aldan on 01/09/2020.
+//  Copyright © 2020 Infinum. All rights reserved.
+//
+
+import XCTest
+import Quick
+import Nimble
+@testable import Catalog
+
+class UserDefaults_PropertyWrapperTests: QuickSpec {
+
+    override func spec() {
```

Comment:
And yet two cases with most logic are not covered :( please test out RawRepresentable init and Codable inits.

Reply:
I have added test cases and minor fixes for Codable and RawRepresentable (non optional). I'll get back to this PR later to add support for optional RawRepresentable

Reply:
As agreed, optional RawRepresentable will not be supported for now :)